datasource/hcp-packer-artifact/data.go

```diff
@@ -0,0 +1,251 @@
+	// Registry when a build is first created, and is unique to this build.
+	BuildID string `mapstructure:"build_id"`
+
+	// The version ID. This is a ULID, which is a unique identifier similar
+	// to a UUID. It is created by the HCP Packer Registry when a version is
+	// first created, and is unique to this version.
+	VersionID string `mapstructure:"version_id"`
+
+	// The ID of the channel used to query the version. This value will be empty if the `version_fingerprint` was used
+	// directly instead of a channel.
+	ChannelID string `mapstructure:"channel_id"`
+
+	// The UUID associated with the Packer run that created this artifact.
+	PackerRunUUID string `mapstructure:"packer_run_uuid"`
+
+	// Identifier or URL of the remote artifact as given by a build.
+	// For example, ami-12345.
+	ExternalIdentifier string `mapstructure:"external_identifier"`
+
+	// The region as given by `packer build`. eg. "ap-east-1".
+	// For locally managed clouds, this may map instead to a cluster, server or datastore.
+	Region string `mapstructure:"region"`
+
+	// The key:value metadata labels associated with this build.
+	Labels map[string]string `mapstructure:"labels"`
+}
+
+func (d *Datasource) OutputSpec() hcldec.ObjectSpec {
+	return (&DatasourceOutput{}).FlatMapstructure().HCL2Spec()
+}
+
+func (d *Datasource) Execute() (cty.Value, error) {
+	ctx := context.TODO()
+
+	cli, err := hcpapi.NewClient()
+	if err != nil {
+		return cty.NullVal(cty.EmptyObject), err
+	}
+
+	var version *hcpPackerModels.HashicorpCloudPacker20230101Version
+	var channelID string
+	if d.config.VersionFingerprint != "" {
+		log.Printf(
+			"[INFO] Reading info from HCP Packer Registry (%s) "+
+				"[project_id=%s, organization_id=%s, version_fingerprint=%s]",
+			d.config.BucketName, cli.ProjectID, cli.OrganizationID, d.config.VersionFingerprint,
+		)
+
+		version, err = cli.GetVersion(ctx, d.config.BucketName, d.config.VersionFingerprint)
+		if err != nil {
+			return cty.NullVal(cty.EmptyObject), fmt.Errorf(
+				"error retrieving version from HCP Packer Registry: %s", err,
+			)
+		}
+	} else {
+		log.Printf(
+			"[INFO] Reading info from HCP Packer Registry (%s) "+
+				"[project_id=%s, organization_id=%s, channel=%s]",
+			d.config.BucketName, cli.ProjectID, cli.OrganizationID, d.config.ChannelName,
+		)
+
+		var channel *hcpPackerModels.HashicorpCloudPacker20230101Channel
+		channel, err = cli.GetChannel(ctx, d.config.BucketName, d.config.ChannelName)
+		if err != nil {
+			return cty.NullVal(cty.EmptyObject), fmt.Errorf(
+				"error retrieving channel from HCP Packer Registry: %s", err.Error(),
+			)
+		}
+
+		if channel.Version == nil {
+			return cty.NullVal(cty.EmptyObject), fmt.Errorf(
+				"there is no version associated with the channel %s", d.config.ChannelName,
+			)
+		}
+		channelID = channel.ID
+		version = channel.Version
+	}
+
+	if version.Status == hcpPackerModels.HashicorpCloudPacker20230101VersionStatusVERSIONREVOKED.Pointer() {
```

Comment:
Calling `.Pointer` actually returns the address of the enum value and not the string value so these will never be equal. What we to do here is deference the value of version.Status and compare it against the enum value. 
```suggestion
	if *version.Status == hcpPackerModels.HashicorpCloudPacker20230101VersionStatusVERSIONREVOKED {
```